File: demo1/lib/models/sensor_data.dart
class SensorData {
  final int heartRate;
  final String latitude;
  final String longitude;
  final String status;
  final DateTime timestamp;

  SensorData({
    required this.heartRate,
    required this.latitude,
    required this.longitude,
    required this.status,
    required this.timestamp,
  });

  factory SensorData.fromJson(Map<String, dynamic> json) {
    return SensorData(
      heartRate: json['heartRate'],
      latitude: json['latitude'],
      longitude: json['longitude'],
      status: json['status'],
      timestamp: DateTime.parse(json['timestamp']),
    );
  }
}

File: models/sensorData.js
const mongoose = require("mongoose");

const sensorDataSchema = new mongoose.Schema({
    heartRate: { type: Number, required: true },
    latitude: { type: String, required: true },
    longitude: { type: String, required: true },
    status: { type: String, required: true },
    timestamp: { type: Date, default: Date.now }
});

module.exports = mongoose.model("SensorData", sensorDataSchema);
